src/XProfile/VirtoCommerce.XProfile/Authorization/CanEditOrganizationAuthorizationHandler.cs

```diff
@@ -53,6 +53,14 @@ protected override async Task HandleRequirementAsync(AuthorizationHandlerContext
                     result = await HasSameOrganizationAsync(currentContact, contactAggregate.Contact.Id);
                 }
             }
+            else if (context.Resource is ApplicationUser applicationUser && currentContact != null)
+            {
+                result = currentContact.Id == applicationUser.MemberId;
```

Comment:
I think you can be limited by only checking `currentUserId  == applicationUser.id` You don't need to use the contact for this resource check.

Reply:
fixed